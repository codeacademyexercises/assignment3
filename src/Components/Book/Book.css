.background1 {
    color: #494748;
    font-weight: 700;
    background-color: #ED965D;
    flex-shrink: 1;
    margin: 20px;
    min-height: 230px;
    min-width: 200px;
    max-width: 265px; /*260*/
    max-height: 290px;  /*285*/
    display: flex;
    flex-direction: column;
    align-items: center;
}

.background2 {
    color: #ED965D;
    font-weight: 700;
    background-color: white;
    flex-shrink: 1;
    margin: 20px;
    min-height: 230px;
    min-width: 200px;
    max-width: 265px;
    max-height: 290px;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.ImageContainer {
    width: 100%;
    height: 55%;
    overflow: hidden;
}

.image{
    width: 100%;
    height: 170%;
}

.BookInfo {
    flex-grow: 1;
    width: 90%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
}

.BookName {
    width: 73%;
    padding-top: 20px;
    text-align: center;
}

/* .BookInfo :last-child{
    align-self: 
} */