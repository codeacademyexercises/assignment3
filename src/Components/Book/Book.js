import React, { Component } from 'react';
import BookReaction from '../../containers/BookReaction';
import './Book.css';

class Book extends Component {
    render(){
        let background='background1';
        if(this.props.index%2===1){
            background='background2';
        }
        return(
            <div className={background}>
                <div className="ImageContainer">
                    <img  className="image" alt='' src={require('../../images/cover_image.jpeg')} />
                </div>
                <div className="BookInfo">
                    <div className="BookName">
                        {this.props.bookinfo.Name.toUpperCase()}
                    </div>
                    <BookReaction bookinfo={this.props.bookinfo} index={this.props.index}/>
                </div>
            </div>
        );
    };
};

export default Book;