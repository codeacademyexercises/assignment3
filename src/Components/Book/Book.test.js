import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Book from './Book';

describe('<Book /> ',()=>{
    it('snapshot Testing',()=>{
        const wrapper = shallow(<Book bookinfo={{Name: 'Harry', rating: 4.5,}}/>);
        expect(wrapper).toMatchSnapshot();
    })
})