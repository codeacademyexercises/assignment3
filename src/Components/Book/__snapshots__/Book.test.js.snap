// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Book />  snapshot Testing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Book
    bookinfo={
      Object {
        "Name": "Harry",
        "rating": 4.5,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          className="image"
          src="cover_image.jpeg"
        />,
        <div
          className="BookInfo"
        >
          Harry
        </div>,
        <div
          className="BookInfo"
        >
          4.5
        </div>,
      ],
      "className": "Book",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": "image",
          "src": "cover_image.jpeg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Harry",
          "className": "BookInfo",
        },
        "ref": null,
        "rendered": "Harry",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 4.5,
          "className": "BookInfo",
        },
        "ref": null,
        "rendered": 4.5,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            className="image"
            src="cover_image.jpeg"
          />,
          <div
            className="BookInfo"
          >
            Harry
          </div>,
          <div
            className="BookInfo"
          >
            4.5
          </div>,
        ],
        "className": "Book",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "image",
            "src": "cover_image.jpeg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Harry",
            "className": "BookInfo",
          },
          "ref": null,
          "rendered": "Harry",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 4.5,
            "className": "BookInfo",
          },
          "ref": null,
          "rendered": 4.5,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
